source/Ship.cpp

```diff
@@ -2257,6 +2257,26 @@ bool Ship::IsOverheated() const
 
 
 
+bool Ship::IsIonized() const
+{
+	if(!ionization)
+		return false;
+
+	// A ship can only be fully ionized if its engines or weapons require energy.
+	bool movementEnergy = attributes.Get("thrusting energy")
```

Comment:
Converting numeric values to bools like this is equivalent to `value != 0`, that is, it is `true` if the value is not zero, and `false` if it is zero. This means that negative values also become `true`, but having negative thrusting energy, for example, doesn't mean energy is required to thrust, it means energy is generated while thrusting. So, these should probably be checking that the energy is actually required and consumed, so for positive values only.